memguard_test.go

```diff
@@ -521,3 +505,43 @@ func TestGetBytes(t *testing.T) {
 		t.Error("pointer does not describe actual memory")
 	}
 }
+
+func TestFinalizer(t *testing.T) {
+	b, err := New(8, false)
+	if err != nil {
+		t.Error("unexpected error")
+	}
+	ib := b.lockedBuffer
```

Comment:
because below I need to call IsDestroyed. if I called b.IsDestroyed b would still be reachable and therefore the finalizer would never be called. By doing ib := b.lockedBuffer (note: lockedBuffer is obviously private!) and calling ib.IsDestroyed instead b is unreachable and therefore the finalizer will be called during the forced GC